### PR TITLE
BC-4570 - Move "FileContentElement" and its subcomponents into separate folder

### PR DESCRIPTION
**Pull request by bergatco**
# Short Description

As with the ongoing implementation the folder `/src/components/feature-board` is getting quite full and somewhat messy. Therefore it was decided to move all files related to the "FileContentElement" and its subcomponent into a separate folder `file-content-element`.

## Links to Ticket and related Pull-Requests

https://ticketsystem.dbildungscloud.de/browse/BC-4570

Link to deployment : https://default-bc-4570-move-file-element-into-folder.cd.dbildungscloud.dev/
Link to board : https://default-bc-4570-move-file-element-into-folder.cd.dbildungscloud.dev/rooms/64b76705564ceeca4d43a122/board

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: src/components/feature-board/content-elements/ContentElementList.vue [hunk 1]
```diff
@@ -31,11 +31,11 @@ import {
 	FileElementResponse,
 	RichTextElementResponse,
 } from "@/serverApi/v3";
-import { computed, defineComponent, PropType } from "vue";
+import { PropType, computed, defineComponent } from "vue";
 import { AnyContentElement } from "../types/ContentElement";
 import { ElementMove } from "../types/DragAndDrop";
-import FileContentElement from "./FileContentElement.vue";
 import RichTextContentElement from "./RichTextContentElement.vue";
+import FileContentElement from "./file-content-element/FileContentElement.vue";
```
Comment (CeEv) [inline src/components/feature-board/content-elements/ContentElementList.vue:38]: Wollen wir mit ein index file arbeiten in dem die Parts stehen die auch außerhalb benötigt werden?

File: src/components/feature-board/content-elements/file-content-element/FileContentElement.unit.ts [hunk 2]
```diff
@@ -10,16 +10,17 @@ import { fileElementResponseFactory } from "@@/tests/test-utils/factory/fileElem
...
 import FileContentElementDisplay from "./FileContentElementDisplay.vue";
 import FileContentElementEdit from "./FileContentElementEdit.vue";
-jest.mock("../shared/InlineEditInteractionHandler.composable");
-jest.mock("../shared/DeleteBoardNodeConfirmation.composable");
-jest.mock("../shared/FileStorageApi.composable");
-jest.mock("../shared/SelectedFile.composable");
+
+jest.mock("../../shared/InlineEditInteractionHandler.composable");
+jest.mock("../../shared/DeleteBoardNodeConfirmation.composable");
+jest.mock("../../shared/FileStorageApi.composable");
```
Comment (CeEv) [inline src/components/feature-board/content-elements/file-content-element/FileContentElement.unit.ts:22]: Ich bin mir bei dem File nicht sicher was drin ist. Ist das wirklich das API Interface? Wenn ja wäre es vermutlich kein composable. Zudem wäre die Frage ob das wirklich etwas ist das shared ist.

File: src/components/feature-board/content-elements/file-content-element/FileContentElement.vue [hunk 3]
```diff
@@ -49,10 +49,10 @@ import {
 } from "@/fileStorageApi/v3";
 import { FileElementResponse } from "@/serverApi/v3";
 import { PropType, computed, defineComponent, onMounted } from "vue";
-import { useDeleteBoardNodeConfirmation } from "../shared/DeleteBoardNodeConfirmation.composable";
-import { useFileStorageApi } from "../shared/FileStorageApi.composable";
-import { useSelectedFile } from "../shared/SelectedFile.composable";
-import { useContentElementState } from "../state/ContentElementState.composable";
+import { useDeleteBoardNodeConfirmation } from "../../shared/DeleteBoardNodeConfirmation.composable";
```
Comment (CeEv) [inline src/components/feature-board/content-elements/file-content-element/FileContentElement.vue:52]: Gibt es ein ref link ala @shared/ ?
Der wäre aus meiner Sicht für *alle* relativen Pfade die außerhalb des eigenenen Bereiches ..in dem Fall file-content-elment folder zu empfehlen.

**Review by CeEv — CHANGES_REQUESTED** [review]

File: src/components/feature-board/content-elements/file-content-element/FileContentElement.vue (hunk 3 above)
Comment (bergatco) [reply src/components/feature-board/content-elements/file-content-element/FileContentElement.vue]: Fixed by replacing relative paths with new aliases!

- Replaced `../../shared/*` with `@boardShared/*` (`src/components/feature-board/shared/*`)
- replaced `../../state/*` with `@boardState/*` (`src/components/feature-board/state/*`)
- replaced `../../types/*` with `@boardTypes/*` (`src/components/feature-board/types/*`)

**HINT! The new aliases needs to be added to `tsconfig.json` as well as `jest.config.js` to be known!!!**

Comment (bergatco) [reply src/components/feature-board/content-elements/file-content-element/FileContentElement.vue]: Applied changes to all files **WITHIN** the new folder, in order to not extend the scope of this PR.

File: src/components/feature-board/content-elements/ContentElementList.vue (hunk 1 above)
Comment (bergatco) [reply src/components/feature-board/content-elements/ContentElementList.vue]: Fixed by adding an index file exporting `FileContentElement` and `useFileRecord` only for now!

File: src/components/feature-board/content-elements/file-content-element/FileContentElement.unit.ts (hunk 2 above)
Comment (bergatco) [reply src/components/feature-board/content-elements/file-content-element/FileContentElement.unit.ts]: Diese Frage sprengt leider den Rahmen dieses Tickets und daher würde ich damit gerne an das Frontend-Arc-Chapter verweisen. Sollten sich hier Änderungen ergeben, würde ich Dich bitte ein neues Ticket zu erstellen.

File: src/components/feature-board/content-elements/ContentElementList.vue [hunk 4]
```diff
@@ -31,11 +31,11 @@ import {
 	FileElementResponse,
 	RichTextElementResponse,
 } from "@/serverApi/v3";
-import { computed, defineComponent, PropType } from "vue";
+import { PropType, computed, defineComponent } from "vue";
```
Comment (MartinSchuhmacher) [inline src/components/feature-board/content-elements/ContentElementList.vue:34]: did you run the linter here and did that one sort the words like that? just irritated why "P" is before "c" and if that might has to do with capital letters

**Review by MartinSchuhmacher — APPROVED** [review]

File: src/components/feature-board/content-elements/ContentElementList.vue (hunk 4 above)
Comment (bergatco) [reply src/components/feature-board/content-elements/ContentElementList.vue:34]: That's correct! In my `.vscode/settings.json` I have enabled `editor.formatOnSave` as well as `source.organizeImports` for the `editor.codeActionsOnSave`. That's why on save the imports are sorted automatically.

```
{
	"[vue]": {
		"editor.defaultFormatter": "esbenp.prettier-vscode"
	},
	"[html]": {
		"editor.defaultFormatter": "esbenp.prettier-vscode"
	},
	"[javascript]": {
		"editor.defaultFormatter": "esbenp.prettier-vscode"
	},
	"[typescript]": {
		"editor.defaultFormatter": "esbenp.prettier-vscode"
	},
	"[json]": {
		"editor.defaultFormatter": "esbenp.prettier-vscode"
	},
	"editor.formatOnSave": true,
	"editor.codeActionsOnSave": {
		"source.organizeImports": true
	},
	"prettier.prettierPath": "",
	"sonarlint.connectedMode.project": {
		"connectionId": "schulcloud-verbund",
		"projectKey": "hpi-schul-cloud_nuxt-client"
	},
}
```

**Review by MartinSchuhmacher — APPROVED** [review]

**Review by MartinSchuhmacher — APPROVED** [review]

**Pull request merged**